restcomm/restcomm.sms/src/main/java/org/restcomm/connect/sms/SmsService.java

```diff
@@ -169,19 +171,64 @@ private void message(final Object message) throws IOException {
         final SipURI fromURI = (SipURI) request.getFrom().getURI();
         final String fromUser = fromURI.getUser();
         final ClientsDao clients = storage.getClientsDao();
+
+        final SipURI ruri = (SipURI) request.getRequestURI();
+        final String to = ruri.getUser();
+
+        Sid destinationOrganizationSid = SIPOrganizationUtil.searchOrganizationBySIPRequest(storage.getOrganizationsDao(), request);
+
         final Sid sourceOrganizationSid = OrganizationUtil.getOrganizationSidBySipURIHost(storage, fromURI);
+
         if(logger.isDebugEnabled()) {
             logger.debug("sourceOrganizationSid: " + sourceOrganizationSid);
+            logger.debug("destinationOrganizationSid: "+destinationOrganizationSid);
         }
         if(sourceOrganizationSid == null){
             logger.error("Null Organization: fromUri: "+fromURI);
         }
+
         final Client client = clients.getClient(fromUser, sourceOrganizationSid);
+        final String toUser = CallControlHelper.getUserSipId(request, useTo);
+        final Client toClient = clients.getClient(toUser, destinationOrganizationSid);
         final AccountsDao accounts = storage.getAccountsDao();
         final ApplicationsDao applications = storage.getApplicationsDao();
 
+        IncomingPhoneNumber number = getIncomingPhoneNumber(to, sourceOrganizationSid, destinationOrganizationSid);
+
+        if (number != null) {
```

Comment:
i notice this piece of code is heavily reuse/duplicated later. can we consider to place this in a common util...?